tests/various/peepopt.ys

```diff
@@ -173,3 +173,34 @@ select -assert-count 1 t:$dff r:WIDTH=2 %i
 select -assert-count 2 t:$mux
 select -assert-count 2 t:$mux r:WIDTH=2 %i
 select -assert-count 0 t:$logic_not t:$dff t:$mux %% t:* %D
+
+####################
+
+design -reset
+read_verilog <<EOT
+module peepopt_dffmuxext_signed_rst_init(input clk, ce, rstn, input signed [1:0] i, output reg signed [3:0] o);
+    initial o <= 4'b0010;
+    always @(posedge clk) begin
+        if (ce) o <= i;
+        if (!rstn) o <= 4'b1111;
+    end
+endmodule
+EOT
+
+proc
+#equiv_opt -assert peepopt
```

Comment:
Is there some deficiency in `equiv_opt` that requires escaping from it here? Or is it just for readability?

In either case, as the author of `equiv_opt`, IMO it should either default to `miter`+`sat` or at least provide it as an option. The reason it does not use those is because I did not understand how to use them at the time. I am still not very certain of this, but I think `miter`+`sat` are strictly more useful than `equiv_induct`. (Correct me if I'm wrong.)

Reply:
Sorry, let me check again. I couldn't get it to fail in equiv_opt and chalked it up to init funniness...

Reply:
That's one more reason to change `equiv_opt` to use `sat`+`miter` then.

Reply:
Okay, I've just force pushed some commits that are easier to work with.

The fix is d99810ad8a0d56a13cca75c4b129f932491d9c80.
Regardless of this commit, `equiv_opt` passes.

Without this commit `miter`+`sat` fails, but *only* with #1427 (i.e. it fails if c6a55d948aebc8993cc3e33372b9c403b0b90554 is reverted).

Fun fun fun!

To be honest, I don't think I fully understand all the pros and cons of each method either, except that `miter`+`sat` is bounded and temporal induction is not, however the latter suffers when (from my own empirical experience) initial conditions or clock enables are involved. From a developer point of view, it would be great to get an absolutely bomb-proof equivalence check for any (sequential or combinatorial) transformation I make...

Reply:
Yeah, agreed on the latter. @cliffordwolf might have a better understanding of this?

Reply:
`sat` has `-tempinduct` for induction, too. I think that `equiv_induct` also adds extra assertions that registers driving the same wire name are equivalent which significantly helps induction, as induction isn't really a useful form of blackbox verification. I don't think `miter` can do this but there might be some other trickery involving the undef modelling in `sat` (although my experience is that comes at a high performance cost)

Reply:
In fact, `sat -tempinduct` looks stronger than `equiv_induct` as it checks the base case too whereas `equiv_induct` appears just to do the induction part.

Reply:
Conversations with @cliffordwolf seem to indicate that `equiv_induct` can/should be augmented to cover the base case (requiring about a day's work) but would that actually help here?

Reply:
It seems like it would. `sat -tempinduct-inductonly` (mirroring `equiv_induct`) passes, as expected, but `sat -tempinduct-baseonly` or `sat -tempinduct` fails which is a good sign. The expectation is that `equiv_induct` that also covers the base case would be a desirable improvement.

Reply:
Adding the base case to `equiv_induct` would be helpful, I think. I wonder if counterexamples would be possible too?